packages/editor/src/utils/badge-properties.ts

```diff
@@ -0,0 +1,27 @@
+import { IvyIcons } from '@axonivy/ui-icons';
+
+const badgePropertyData = {
+  regex: /#{\s*data\.[^}]+}/,
+  icon: IvyIcons.File,
+  badgeTextGen: (text: string) => text.replaceAll(/#{\s*data\.|}/g, '')
+};
```

Comment:
The icons used for the badges are still up for debate.
````typescript
  icon: IvyIcons.File,
  icon: IvyIcons.Bpmn,
  icon: IvyIcons.DataClass,
  icon: IvyIcons.Cms,
````

Reply:
For now, I’d use the same icons as the Browser: `IvyIcons.Attribute` for badgePropertyData and I`vyIcons.Process` for badgePropertyLogic. The other badges should already have the correct icons. But maybe there are better options, or Anais could create new ones.